pandas/_testing/asserters.py

```diff
@@ -914,8 +920,23 @@ def assert_series_equal(
             or is_numeric_dtype(right.dtype)
             and not is_float_dtype(right.dtype)
         )
+        left_index_dtypes = (
+            [left.index.dtype] if left.index.nlevels == 1 else left.index.dtypes
+        )
+        right_index_dtypes = (
+            [right.index.dtype] if right.index.nlevels == 1 else right.index.dtypes
+        )
+        check_exact_index = (
+            all(is_numeric_dtype(dtype) for dtype in left_index_dtypes)
+            and not any(is_float_dtype(dtype) for dtype in left_index_dtypes)
```

Comment:
You could check `dtype.kind in "iu"` to check integer types in one go

Reply:
Brilliant! That doesn't seem to work for multiindexes though as the dtype is always object (why?)